src/jdk.compiler/share/classes/com/sun/tools/javac/comp/ThisEscapeAnalyzer.java

```diff
@@ -442,25 +442,6 @@ public void scan(JCTree tree) {
 
         // Sanity check
         Assert.check(checkInvariants(true, referenceExpressionNode));
```

Comment:
Just a note that now the only reason we set up that big switch is to capture `referenceExpressionNode` which is only used in `checkInvariants` (which looks more of a debugging function). I'll leave it up to you whether to keep it as is.

Reply:
Correct... the debug check is valuable and low cost so I think it's worth keeping for now.